test/model/message_list_test.dart

```diff
@@ -1703,6 +1703,23 @@ void main() {
         }
       }
     });
+    test('stream messages match with case-insensitive topics', () {
```

Comment:
nit: put this new test just under the test 'stream messages match just if same stream/topic'

Reply:
Also here's a version that has different tests for each case (instead of multiple checks in the same test), along with a descriptive label for each test, which will appear in output when it fails. This also tests some additional cases that are interesting:

```dart
    group('topics compared case-insensitively', () {
      void doTest(String description, String topicA, String topicB, bool expected) {
        test(description, () {
          final stream = eg.stream();
          final messageA = eg.streamMessage(stream: stream, topic: topicA);
          final messageB = eg.streamMessage(stream: stream, topic: topicB);
          check(haveSameRecipient(messageA, messageB)).equals(expected);
        });
      }

      doTest('same case, all lower',               'abc',  'abc',  true);
      doTest('same case, all upper',               'ABC',  'ABC',  true);
      doTest('same case, mixed',                   'AbC',  'AbC',  true);
      doTest('same non-letter chars',              '嗎',    '嗎',    true);
      doTest('different case',                     'aBc',  'ABC',  true);
      doTest('different case, same diacritics',    'AbÇ',  'aBç',  true);
      doTest('same letters, different diacritics', 'ma',   'mǎ',   false);
      doTest('having different non-letter chars',  'ma 嗎', 'mǎ 馬', false);
    });
```

Reply:
Yes, Thanks for the help. This test looks promising and checks for each case. I'll just replaces this with mine.